{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "mount_file_id": "1k3fWPvuXj4YU2TK5c7gfM_YXmwrogsgR",
      "authorship_tag": "ABX9TyOVhV5VnJCCE1qRY4SOHeLn",
      "include_colab_link": true
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/Rafaelca70x/estagio-bio-info-LGHM/blob/main/genomica_popul_3.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import pandas as pd\n",
        "import numpy as np\n",
        "import matplotlib.pyplot as plt\n",
        "import seaborn as sns"
      ],
      "metadata": {
        "id": "vWqXYBk_bF8-"
      },
      "execution_count": 1,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#localização do CFTR no chr7: 117465784 - 117715971"
      ],
      "metadata": {
        "id": "Vs_qhhhgtJhx"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "_______________________________________________________________________________________\n",
        "- descrever o CFTR\n",
        "- as variantes do GNOMAD (quantas variantes são raras, quantas são comuns - frequencia alelica)\n",
        "- distribuição das patogenicidades\n",
        "- relacionar com as LOFs\n",
        "- implicação para a população miscigenada (latino/admix american)\n"
      ],
      "metadata": {
        "id": "q2mEwV4AgY0t"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "- **Allele count** in genotypes for each ALT allele, in the same order as listed\n",
        "- Total **number** of alleles in called genotypes\n",
        "- ex:\n",
        "   - Let's assume the genotype 0/1, 1/1 and 1/2. AC is 4,1 (4 times ALT allele 1 and 1 time ALT allele 2) and AN is 6."
      ],
      "metadata": {
        "id": "oPtgjpOZxD1L"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "_____________________________________________________________________________________________\n",
        "- comparar as posições em comum nos dados do gnomad e do CFTR_br"
      ],
      "metadata": {
        "id": "J9oZ-Nh_tVuE"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "path_cftr_br = r'/content/drive/MyDrive/Colab Notebooks/ESTAGIO LGHM/CFTR_BR_freq.csv'\n",
        "cftr_br = pd.read_csv(path_cftr_br, sep = ',', low_memory=False) #frequencia alelica - 22 variantes"
      ],
      "metadata": {
        "id": "M1VoqXcmtSPl"
      },
      "execution_count": 2,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "gnomad_cftr_v3_path = r'/content/drive/MyDrive/Colab Notebooks/ESTAGIO LGHM/gnomAD_v3.1.2.csv' #versão 3\n",
        "gnomad_cftr_v3 = pd.read_csv(gnomad_cftr_v3_path)"
      ],
      "metadata": {
        "id": "Vdb-H73Fa9-Z"
      },
      "execution_count": 3,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pos_com = pd.Series(np.intersect1d(cftr_br.POS, gnomad_cftr_v3.Position)) #retorna as posições interseccionadas do CFTR_BR e do CFTR do gnomad"
      ],
      "metadata": {
        "id": "D8Bg8QZsiOz7"
      },
      "execution_count": 59,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "pos_df = pd.concat([cftr_br['POS'],pos_com],axis=1,sort=True)\n",
        "pos_df.rename(columns = {'POS': 'POS_CFTR_BR', 0:'POS_GNOMAD_V3'}, inplace=True)"
      ],
      "metadata": {
        "id": "LMYLctddgF2W"
      },
      "execution_count": null,
      "outputs": []
    },
    {
      "cell_type": "code",
      "source": [
        "#posição faltante no CFTR do gnomad_v3\n",
        "set(pos_df.POS_CFTR_BR) - set(pos_df.POS_GNOMAD_V3)\n",
        "#pos_df['POS_CFTR_BR'].isin(pos_df['POS_GNOMAD_V3'])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "nmUluF1nXGK5",
        "outputId": "0bbf0e7e-4722-4d81-b2ea-fe0f2f587b4a"
      },
      "execution_count": null,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{117559606}]"
            ]
          },
          "metadata": {},
          "execution_count": 22
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "____________________________________________________________________\n",
        "- distribuição das patogenicidades\n",
        "     - **VEP Annotation**: Variant Effect Predictor for most severe consequences across all transcripts for this gene\n",
        "     - **ClinVar Clinical Significance**"
      ],
      "metadata": {
        "id": "boCD9Dop0YxE"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "pd.crosstab(gnomad_cftr_v3['VEP Annotation'], gnomad_cftr_v3['ClinVar Clinical Significance'])"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 523
        },
        "id": "rJ-QXhOL16hT",
        "outputId": "862d17df-aff5-4337-f487-319c9091b980"
      },
      "execution_count": 57,
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "ClinVar Clinical Significance  Benign  Benign/Likely benign  \\\n",
              "VEP Annotation                                                \n",
              "3_prime_UTR_variant                 0                     1   \n",
              "5_prime_UTR_variant                 0                     2   \n",
              "frameshift_variant                  0                     0   \n",
              "inframe_deletion                    0                     0   \n",
              "inframe_insertion                   0                     0   \n",
              "intron_variant                     19                    11   \n",
              "missense_variant                    0                     2   \n",
              "splice_acceptor_variant             0                     0   \n",
              "splice_donor_variant                0                     0   \n",
              "splice_region_variant               0                     1   \n",
              "start_lost                          0                     0   \n",
              "stop_gained                         0                     0   \n",
              "synonymous_variant                  1                     7   \n",
              "\n",
              "ClinVar Clinical Significance  Conflicting interpretations of pathogenicity  \\\n",
              "VEP Annotation                                                                \n",
              "3_prime_UTR_variant                                                       1   \n",
              "5_prime_UTR_variant                                                       0   \n",
              "frameshift_variant                                                        1   \n",
              "inframe_deletion                                                          0   \n",
              "inframe_insertion                                                         0   \n",
              "intron_variant                                                           34   \n",
              "missense_variant                                                        114   \n",
              "splice_acceptor_variant                                                   0   \n",
              "splice_donor_variant                                                      1   \n",
              "splice_region_variant                                                    17   \n",
              "start_lost                                                                0   \n",
              "stop_gained                                                               1   \n",
              "synonymous_variant                                                       44   \n",
              "\n",
              "ClinVar Clinical Significance  Conflicting interpretations of pathogenicity; other  \\\n",
              "VEP Annotation                                                                       \n",
              "3_prime_UTR_variant                                                            0     \n",
              "5_prime_UTR_variant                                                            0     \n",
              "frameshift_variant                                                             0     \n",
              "inframe_deletion                                                               0     \n",
              "inframe_insertion                                                              0     \n",
              "intron_variant                                                                 1     \n",
              "missense_variant                                                               0     \n",
              "splice_acceptor_variant                                                        0     \n",
              "splice_donor_variant                                                           0     \n",
              "splice_region_variant                                                          0     \n",
              "start_lost                                                                     0     \n",
              "stop_gained                                                                    0     \n",
              "synonymous_variant                                                             0     \n",
              "\n",
              "ClinVar Clinical Significance  Likely benign  Likely pathogenic  Pathogenic  \\\n",
              "VEP Annotation                                                                \n",
              "3_prime_UTR_variant                        0                  0           0   \n",
              "5_prime_UTR_variant                        0                  0           0   \n",
              "frameshift_variant                         0                  0          28   \n",
              "inframe_deletion                           0                  0           4   \n",
              "inframe_insertion                          0                  0           1   \n",
              "intron_variant                            40                  0           1   \n",
              "missense_variant                           2                  7          40   \n",
              "splice_acceptor_variant                    0                  0           4   \n",
              "splice_donor_variant                       0                  0          11   \n",
              "splice_region_variant                     12                  1           7   \n",
              "start_lost                                 0                  0           1   \n",
              "stop_gained                                0                  2          26   \n",
              "synonymous_variant                       129                  0           0   \n",
              "\n",
              "ClinVar Clinical Significance  Pathogenic/Likely pathogenic  \\\n",
              "VEP Annotation                                                \n",
              "3_prime_UTR_variant                                       0   \n",
              "5_prime_UTR_variant                                       0   \n",
              "frameshift_variant                                        2   \n",
              "inframe_deletion                                          1   \n",
              "inframe_insertion                                         0   \n",
              "intron_variant                                            0   \n",
              "missense_variant                                          5   \n",
              "splice_acceptor_variant                                   1   \n",
              "splice_donor_variant                                      1   \n",
              "splice_region_variant                                     0   \n",
              "start_lost                                                0   \n",
              "stop_gained                                               1   \n",
              "synonymous_variant                                        0   \n",
              "\n",
              "ClinVar Clinical Significance  Pathogenic; drug response  \\\n",
              "VEP Annotation                                             \n",
              "3_prime_UTR_variant                                    0   \n",
              "5_prime_UTR_variant                                    0   \n",
              "frameshift_variant                                     0   \n",
              "inframe_deletion                                       0   \n",
              "inframe_insertion                                      0   \n",
              "intron_variant                                         0   \n",
              "missense_variant                                       8   \n",
              "splice_acceptor_variant                                0   \n",
              "splice_donor_variant                                   0   \n",
              "splice_region_variant                                  0   \n",
              "start_lost                                             0   \n",
              "stop_gained                                            0   \n",
              "synonymous_variant                                     0   \n",
              "\n",
              "ClinVar Clinical Significance  Uncertain significance  drug response  \\\n",
              "VEP Annotation                                                         \n",
              "3_prime_UTR_variant                                 1              0   \n",
              "5_prime_UTR_variant                                 0              0   \n",
              "frameshift_variant                                  0              0   \n",
              "inframe_deletion                                    0              0   \n",
              "inframe_insertion                                   0              0   \n",
              "intron_variant                                     22              0   \n",
              "missense_variant                                  368              8   \n",
              "splice_acceptor_variant                             0              0   \n",
              "splice_donor_variant                                0              0   \n",
              "splice_region_variant                              19              0   \n",
              "start_lost                                          1              0   \n",
              "stop_gained                                         0              0   \n",
              "synonymous_variant                                  1              0   \n",
              "\n",
              "ClinVar Clinical Significance  not provided  \n",
              "VEP Annotation                               \n",
              "3_prime_UTR_variant                       0  \n",
              "5_prime_UTR_variant                       0  \n",
              "frameshift_variant                        0  \n",
              "inframe_deletion                          1  \n",
              "inframe_insertion                         0  \n",
              "intron_variant                            0  \n",
              "missense_variant                          3  \n",
              "splice_acceptor_variant                   0  \n",
              "splice_donor_variant                      1  \n",
              "splice_region_variant                     1  \n",
              "start_lost                                0  \n",
              "stop_gained                               0  \n",
              "synonymous_variant                        0  "
            ],
            "text/html": [
              "\n",
              "  <div id=\"df-ec0b171c-44dc-45e9-8c3e-8f30c5a6a948\" class=\"colab-df-container\">\n",
              "    <div>\n",
              "<style scoped>\n",
              "    .dataframe tbody tr th:only-of-type {\n",
              "        vertical-align: middle;\n",
              "    }\n",
              "\n",
              "    .dataframe tbody tr th {\n",
              "        vertical-align: top;\n",
              "    }\n",
              "\n",
              "    .dataframe thead th {\n",
              "        text-align: right;\n",
              "    }\n",
              "</style>\n",
              "<table border=\"1\" class=\"dataframe\">\n",
              "  <thead>\n",
              "    <tr style=\"text-align: right;\">\n",
              "      <th>ClinVar Clinical Significance</th>\n",
              "      <th>Benign</th>\n",
              "      <th>Benign/Likely benign</th>\n",
              "      <th>Conflicting interpretations of pathogenicity</th>\n",
              "      <th>Conflicting interpretations of pathogenicity; other</th>\n",
              "      <th>Likely benign</th>\n",
              "      <th>Likely pathogenic</th>\n",
              "      <th>Pathogenic</th>\n",
              "      <th>Pathogenic/Likely pathogenic</th>\n",
              "      <th>Pathogenic; drug response</th>\n",
              "      <th>Uncertain significance</th>\n",
              "      <th>drug response</th>\n",
              "      <th>not provided</th>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>VEP Annotation</th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "      <th></th>\n",
              "    </tr>\n",
              "  </thead>\n",
              "  <tbody>\n",
              "    <tr>\n",
              "      <th>3_prime_UTR_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>5_prime_UTR_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>frameshift_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>28</td>\n",
              "      <td>2</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>inframe_deletion</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>4</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>inframe_insertion</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>intron_variant</th>\n",
              "      <td>19</td>\n",
              "      <td>11</td>\n",
              "      <td>34</td>\n",
              "      <td>1</td>\n",
              "      <td>40</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>22</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>missense_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>114</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>7</td>\n",
              "      <td>40</td>\n",
              "      <td>5</td>\n",
              "      <td>8</td>\n",
              "      <td>368</td>\n",
              "      <td>8</td>\n",
              "      <td>3</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>splice_acceptor_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>4</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>splice_donor_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>11</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>splice_region_variant</th>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>17</td>\n",
              "      <td>0</td>\n",
              "      <td>12</td>\n",
              "      <td>1</td>\n",
              "      <td>7</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>19</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>start_lost</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>stop_gained</th>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>2</td>\n",
              "      <td>26</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "    <tr>\n",
              "      <th>synonymous_variant</th>\n",
              "      <td>1</td>\n",
              "      <td>7</td>\n",
              "      <td>44</td>\n",
              "      <td>0</td>\n",
              "      <td>129</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "      <td>1</td>\n",
              "      <td>0</td>\n",
              "      <td>0</td>\n",
              "    </tr>\n",
              "  </tbody>\n",
              "</table>\n",
              "</div>\n",
              "    <div class=\"colab-df-buttons\">\n",
              "\n",
              "  <div class=\"colab-df-container\">\n",
              "    <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-ec0b171c-44dc-45e9-8c3e-8f30c5a6a948')\"\n",
              "            title=\"Convert this dataframe to an interactive table.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\" viewBox=\"0 -960 960 960\">\n",
              "    <path d=\"M120-120v-720h720v720H120Zm60-500h600v-160H180v160Zm220 220h160v-160H400v160Zm0 220h160v-160H400v160ZM180-400h160v-160H180v160Zm440 0h160v-160H620v160ZM180-180h160v-160H180v160Zm440 0h160v-160H620v160Z\"/>\n",
              "  </svg>\n",
              "    </button>\n",
              "\n",
              "  <style>\n",
              "    .colab-df-container {\n",
              "      display:flex;\n",
              "      gap: 12px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert {\n",
              "      background-color: #E8F0FE;\n",
              "      border: none;\n",
              "      border-radius: 50%;\n",
              "      cursor: pointer;\n",
              "      display: none;\n",
              "      fill: #1967D2;\n",
              "      height: 32px;\n",
              "      padding: 0 0 0 0;\n",
              "      width: 32px;\n",
              "    }\n",
              "\n",
              "    .colab-df-convert:hover {\n",
              "      background-color: #E2EBFA;\n",
              "      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "      fill: #174EA6;\n",
              "    }\n",
              "\n",
              "    .colab-df-buttons div {\n",
              "      margin-bottom: 4px;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert {\n",
              "      background-color: #3B4455;\n",
              "      fill: #D2E3FC;\n",
              "    }\n",
              "\n",
              "    [theme=dark] .colab-df-convert:hover {\n",
              "      background-color: #434B5C;\n",
              "      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "      fill: #FFFFFF;\n",
              "    }\n",
              "  </style>\n",
              "\n",
              "    <script>\n",
              "      const buttonEl =\n",
              "        document.querySelector('#df-ec0b171c-44dc-45e9-8c3e-8f30c5a6a948 button.colab-df-convert');\n",
              "      buttonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "\n",
              "      async function convertToInteractive(key) {\n",
              "        const element = document.querySelector('#df-ec0b171c-44dc-45e9-8c3e-8f30c5a6a948');\n",
              "        const dataTable =\n",
              "          await google.colab.kernel.invokeFunction('convertToInteractive',\n",
              "                                                    [key], {});\n",
              "        if (!dataTable) return;\n",
              "\n",
              "        const docLinkHtml = 'Like what you see? Visit the ' +\n",
              "          '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n",
              "          + ' to learn more about interactive tables.';\n",
              "        element.innerHTML = '';\n",
              "        dataTable['output_type'] = 'display_data';\n",
              "        await google.colab.output.renderOutput(dataTable, element);\n",
              "        const docLink = document.createElement('div');\n",
              "        docLink.innerHTML = docLinkHtml;\n",
              "        element.appendChild(docLink);\n",
              "      }\n",
              "    </script>\n",
              "  </div>\n",
              "\n",
              "\n",
              "<div id=\"df-8871d4f4-df75-491b-83c9-19802e7a2d2c\">\n",
              "  <button class=\"colab-df-quickchart\" onclick=\"quickchart('df-8871d4f4-df75-491b-83c9-19802e7a2d2c')\"\n",
              "            title=\"Suggest charts.\"\n",
              "            style=\"display:none;\">\n",
              "\n",
              "<svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n",
              "     width=\"24px\">\n",
              "    <g>\n",
              "        <path d=\"M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z\"/>\n",
              "    </g>\n",
              "</svg>\n",
              "  </button>\n",
              "\n",
              "<style>\n",
              "  .colab-df-quickchart {\n",
              "    background-color: #E8F0FE;\n",
              "    border: none;\n",
              "    border-radius: 50%;\n",
              "    cursor: pointer;\n",
              "    display: none;\n",
              "    fill: #1967D2;\n",
              "    height: 32px;\n",
              "    padding: 0 0 0 0;\n",
              "    width: 32px;\n",
              "  }\n",
              "\n",
              "  .colab-df-quickchart:hover {\n",
              "    background-color: #E2EBFA;\n",
              "    box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n",
              "    fill: #174EA6;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart {\n",
              "    background-color: #3B4455;\n",
              "    fill: #D2E3FC;\n",
              "  }\n",
              "\n",
              "  [theme=dark] .colab-df-quickchart:hover {\n",
              "    background-color: #434B5C;\n",
              "    box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n",
              "    filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n",
              "    fill: #FFFFFF;\n",
              "  }\n",
              "</style>\n",
              "\n",
              "  <script>\n",
              "    async function quickchart(key) {\n",
              "      const charts = await google.colab.kernel.invokeFunction(\n",
              "          'suggestCharts', [key], {});\n",
              "    }\n",
              "    (() => {\n",
              "      let quickchartButtonEl =\n",
              "        document.querySelector('#df-8871d4f4-df75-491b-83c9-19802e7a2d2c button');\n",
              "      quickchartButtonEl.style.display =\n",
              "        google.colab.kernel.accessAllowed ? 'block' : 'none';\n",
              "    })();\n",
              "  </script>\n",
              "</div>\n",
              "    </div>\n",
              "  </div>\n"
            ]
          },
          "metadata": {},
          "execution_count": 57
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [],
      "metadata": {
        "id": "4wUWdCwF9Td8"
      },
      "execution_count": null,
      "outputs": []
    }
  ]
}